### PR TITLE
statistics: update for py310

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: stdlib/statistics.pyi [hunk 1]
```diff
@@ -18,7 +19,13 @@ if sys.version_info >= (3, 8):
     def geometric_mean(data: Iterable[SupportsFloat]) -> float: ...
 
 def mean(data: Iterable[_NumberT]) -> _NumberT: ...
-def harmonic_mean(data: Iterable[_NumberT]) -> _NumberT: ...
+
+if sys.version_info >= (3, 10):
+    def harmonic_mean(data: Iterable[_NumberT], weights: Optional[Iterable[_Number]] = ...) -> _NumberT: ...
```
Comment (JelleZijlstra) [inline stdlib/statistics.pyi:24]: I think this should also be `_NumberT`; `statistics.harmonic_mean([1, 2.0], weights=[Decimal(3), Decimal(4)])` throws.

File: stdlib/statistics.pyi [hunk 2]
```diff
@@ -70,3 +77,11 @@ if sys.version_info >= (3, 8):
         def __rsub__(self, x2: Union[float, NormalDist]) -> NormalDist: ...
         __rmul__ = __mul__
         def __hash__(self) -> int: ...
+
+if sys.version_info >= (3, 10):
+    def correlation(__x: Sequence[_Number], __y: Sequence[_Number]) -> float: ...
```
Comment (JelleZijlstra) [inline stdlib/statistics.pyi:82]: correlation and linear_regression also throw when mixing different types, but covariance doesn't somehow, at least for the example I tried. That means at least the first two should also use `_NumberT`.

**Comment by hauntsaninja** [comment]
Oh, interesting. I tried things with Fraction and it worked out... I guess in the spirit of no false positives, we should keep this as is?

**Comment by JelleZijlstra** [comment]
For what it's worth, mixing Fractions and floats also works for some existing functions that are using the TypeVar:
```
>>> statistics.median([0.5, Fraction(1, 2)])
0.5
```

**Pull request merged**